File: task-03/C++/prime_number.cpp
#include <iostream>
#include <cmath>

using namespace std; 

int main() {
    int n;
    cout << "Enter the limit: ";
    cin >> n;

    if (n < 2) {
        cout << "No Prime" << endl;
        return 0;
    }

    cout << "The prime numbers are:";
    for (int i = 2; i <= n; i++) {
        bool isPrime = true;
        for (int j = 2; j <= sqrt(i); j++) {
            if (i % j == 0) {
                isPrime = false;
                break;
            }
        }
        if (isPrime) {
            cout << " " << i;
        }
    }
    cout << endl;

    return 0;
}

File: task-03/Elixir/prime_number.ex
defmodule PrimeNumbers do
  def main do
    IO.gets("Enter the limit: ") |> String.trim() |> String.to_integer() |> find_primes()
  end

  defp find_primes(n) when n < 2 do
    IO.puts("No Prime")
  end

  defp find_primes(n) do
    IO.write("The prime numbers are:")
    find_primes(2, n)
    IO.puts()
  end

  defp find_primes(n, n) do
    case is_prime(n, 2) do
      true -> IO.write(" #{n}")
      false -> :ok
    end
  end

  defp find_primes(i, n) do
    case is_prime(i, 2) do
      true -> IO.write(" #{i}")
      false -> :ok
    end
    find_primes(i + 1, n)
  end

  defp is_prime(2,2), do: true
  defp is_prime(n, 2) when rem(n,2) == 0, do: false
  defp is_prime(n, i) when i * i > n, do: true
  defp is_prime(n, i) when rem(n,i) == 0, do: false
  defp is_prime(n, i) do
    is_prime(n, i + 1)
  end
end

PrimeNumbers.main()
